packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/gux-rich-text-editor-action/gux-rich-text-editor-action-rich-style/gux-rich-text-editor-action-rich-style.tsx

```diff
@@ -0,0 +1,184 @@
+
+    switch (event.key) {
+      case 'Escape':
+        if (isListEvent) {
+          event.preventDefault();
+          this.expanded = false;
+          this.actionButton.focus();
+        }
+        break;
+      case 'ArrowDown':
+      case 'Enter': {
+        if (isButtonEvent && !this.expanded) {
+          event.preventDefault();
+          this.expanded = true;
+          this.focusFirstListItem();
+        }
+        break;
+      }
+      case 'Tab':
+        this.expanded = false;
+        break;
+      case 'ArrowUp':
+        if (isButtonEvent && !this.expanded) {
+          event.preventDefault();
+          this.expanded = true;
+          this.focusLastListItem();
+        }
+    }
+  }
+
+  @Listen('keyup')
+  handleKeyup(event: KeyboardEvent): void {
+    switch (event.key) {
+      case ' ': {
+        if (event.composedPath().includes(this.actionButton)) {
+          event.preventDefault();
+          this.expanded = true;
+          this.focusFirstListItem();
+        }
+        break;
+      }
+    }
+  }
+
+  private focusFirstListItem(): void {
+    afterNextRender(() => {
+      void this.listElement.guxFocusFirstItem();
+    });
+  }
+
+  private focusLastListItem(): void {
+    afterNextRender(() => {
+      void this.listElement.guxFocusLastItem();
+    });
+  }
+
+  async componentWillLoad(): Promise<void> {
+    trackComponent(this.root);
+    this.i18n = await buildI18nForComponent(this.root, translationResources);
+    this.disabled = calculateDisabledState(this.root);
+  }
+
+  private onActionButtonClick(): void {
+    this.expanded = !this.expanded;
+    if (this.expanded) {
+      this.focusFirstListItem();
+    }
+  }
+
+  private onListClick(event: MouseEvent): void {
+    whenEventIsFrom('gux-rich-style-list-item', event, () => {
+      this.expanded = false;
+      this.actionButton.focus();
+    });
+  }
+
+  private renderPopup(): JSX.Element {
+    return (
+      <div class="gux-list-container" slot="popup">
+        <gux-rich-text-editor-list
+          ref={el => (this.listElement = el)}
+          onClick={e => this.onListClick(e)}
+        >
+          <slot />
+        </gux-rich-text-editor-list>
+      </div>
+    ) as JSX.Element;
+  }
+
+  private renderTooltip(): JSX.Element {
+    if (!this.disabled && !this.expanded) {
+      return (
+        <gux-tooltip-beta>
+          <div slot="content">{this.i18n('richStyle')}</div>
+        </gux-tooltip-beta>
+      ) as JSX.Element;
+    }
+  }
+
+  private renderTarget(): JSX.Element {
+    return (
+      <gux-button-slot accent="ghost" slot="target">
+        <button
+          type="button"
+          ref={el => (this.actionButton = el)}
+          class={{ 'gux-is-active': this.expanded }}
+          onClick={() => this.onActionButtonClick()}
+          disabled={this.disabled}
+          aria-haspopup="true"
+          aria-expanded={this.expanded.toString()}
+        >
+          {this.i18n('paragraph')}
```

Comment:
Going to confirm with UX if this needs to be dynamic depending on the text-style selected.